debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/J9ObjectFieldOffsetIterator_V1.java

```diff
@@ -214,7 +215,20 @@ private void fieldOffsetsFindNext() throws CorruptDataException {
 							J9ClassPointer fieldClass = valueTypeHelper.findJ9ClassInFlattenedClassCacheWithFieldName(instanceClass, J9ROMFieldShapeHelper.getName(localField));
 							if (valueTypeHelper.isJ9FieldIsFlattened(fieldClass, localField)) {
 								UDATA size = null;
-								if (valueTypeHelper.isJ9ClassLargestAlignmentConstraintDouble(fieldClass)) {
+								boolean forceDoudbleAlignment = false;
+								if (fj9object_t_SizeOf == U32.SIZEOF) {
```

Comment:
Perhaps this should use `J9ObjectHelper.compressObjectReferences` directly?

Reply:
We need to cover the case of 32-bit platforms. We can switch to use `J9ObjectHelper.compressObjectReferences` if we won't build valhalla on 32-bit platforms.

Reply:
`ObjectFieldInfo` says:
```
  public static final int fj9object_t_SizeOf =
      (J9ObjectHelper.compressObjectReferences ? U32.SIZEOF : UDATA.SIZEOF);
```
I don't see how replacing `ObjectFieldInfo.fj9object_t_SizeOf == U32.SIZEOF` with `J9ObjectHelper.compressObjectReferences` could be problematic.

Reply:
On 32-bit platforms, `ObjectFieldInfo.fj9object_t_SizeOf` is 4 and `U32.SIZEOF` is also 4. `ObjectFieldInfo.fj9object_t_SizeOf == U32.SIZEOF` is true.

So you are saying that `J9ObjectHelper.compressObjectReferences` will always be true on 32-bit platforms ?

Reply:
No it will never be on. The code is correct as it stands.

Reply:
I stand corrected. I should have looked further and noted that when `J9ObjectHelper.compressObjectReferences` is false, it might be true that `UDATA.SIZEOF == U32.SIZEOF` (i.e. on 32-bit platforms). Sorry for the churn.